src/common/search/SearchPlays.jsx

```diff
@@ -23,11 +23,9 @@ const SearchPlays = () => {
 
   const handleSearch = (event) => {
     event.preventDefault();
-    if (event.key === "Enter") {
-      setSearchTerm(event.target.value);
-      if (location.pathname !== "/plays") {
-        navigate("/plays", { replace: true });
-      }
+    setSearchTerm(event.target.value);
```

Comment:
@Programming-School-Pro-Coding @hasnainmakada-99 

I have a few questions to get answered before we accept the implementation.

- We are moving the play meta data to DB. It means this meta-information will not be available on the client side all the time. How effective it would be to make an API call on every key entered?
- Ok we may assume, all the play records we will pull from an API call on the first render and keep in the memory to perform the type-ahead search. That's acceptable but will be limited. When the number of plays grows say 100 or more(now it is ~30), it will not be effective to fetch all the play info at once. Hence, we want to bring a lazy load on scroll/some sort of pagination. In this case, the play you are searching for will not be available unless we make an API call and find it based on the search term. It will be expensive to on every key enter. What say?
- The playlist page today is getting cluttered with boxes. We will be revamping this page soon to have featured plays at the top, then plays categorized, etc. How does my above concern fit with respect to thin implementation?

I am asking so many things ahead so that we don't have to redo anything later. Otherwise, the code changes look good to me.

Reply:
Yes @atapas sir you are absolutely right in we also have to look the scalability issue too, I agree with you so lets put this as it is (The enter key version) what do you say @atapas sir

Reply:
Tapas sir should I close this issue now  @atapas

Reply:
This would make sense for a client-side filter with on play-list which is already loaded. While going to DB, it will be having performance issues even though we build a caching mechanism as the number of play will only grow. 

I think we can still have a client-side filter/sorting mechanism however, note that, it will only be acting on already loaded plays.

Reply:
Yes @koustov I agree with you we already have a filter mechanism on client side so, and also looking from the database side it will raise performance issues too if we apply this functionality, So its better for all of us that we should not implement this, Am I right ? @atapas sir

Reply:
I agree with all of you.

Reply:
So should I close this Pr and issue now

Reply:
I agree @hasnainmakada-99 

It was a good attempt to fix by @Programming-School-Pro-Coding but we do not need the fix due to above-stated reasons.

Reply:
Ok, I will close the pull request